scalalib/src/mill/javalib/android/AndroidAppModule.scala

```diff
@@ -47,6 +48,15 @@ object AndroidLintReportFormat extends Enumeration {
 @mill.api.experimental
 trait AndroidAppModule extends JavaModule {
 
+  final def root: Path = super.millSourcePath
+  private def src: Path = root / "src"
+  override def millSourcePath: Path = src / "main"
+
+  override def sources: T[Seq[PathRef]] = Task.Sources(millSourcePath / "java")
```

Comment:
If we want to follow the `src/main/java/` etc. convention, rather than configuring it ourselves we should be able to mix in `with MavenModule` to set that config

Reply:
If we want to make `AndroidAppModule` have the `MavenModule` layout by default, we should introduce another trait (maybe `AndroidAppBasicModule`???) that keeps the current layout, to preserve the symmetry with `JavaModule`/`MavenModule`, `KotlinModule`/`KotlinMavenModule`, `ScalaModule`/`SbtModule`

Reply:
I have to check if it's what Android follows and makes sense, after you mentioned it I bet it is! but if it's not, maybe we are coupling the android structure with the maven one, I need more time to check.

The android structure is that it has the java directory that hosts both kotlin and java classes . Maybe we'll want to support scala too, it could be a good idea to not couple them for now, but up to you

Reply:
I gave it a go with MavenModule but it entangled the paths even further and I gave up for now. Maybe we can consolidate later once we have basic working scenarios and tests

Reply:
Sounds good, we can just go with this for now